### ISSUE TITLE
[BUG] mlflow.pytorch.autlog() not working for params

### ISSUE THREAD

**Issue by abhiksark** [mlflow/mlflow#4177]
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [X] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
- **Linux Ubuntu 18.04)**:
- **MLflow installed from source**:
- **MLflow version: 1.13.1**:
- **Python version: 3.8.5 **:
-**Pytorch version: 1.7.1 **: 

### Describe the problem
I was using `mlflow.pytorch.autolog(log_every_n_epoch=1, log_models=False, disable=True, exclusive=False)` to log Pytorch run. But I ran into an issue where it was not logging any of my metrics and parameters. 

### Code to reproduce issue
    with mlflow.start_run() as run:
        mlflow.pytorch.autolog(log_every_n_epoch=1, log_models=False, disable=True, exclusive=False)    
        for epoch in tqdm(range(EPOCHS)):
            start_time = time.monotonic()

            train_loss, train_acc_1, train_acc_5 = train(
            model, train_iterator, optimizer, criterion, scheduler, device)
            valid_loss, valid_acc_1, valid_acc_5 = evaluate(
            model, valid_iterator, criterion, device)

            if valid_loss < best_valid_loss:
                best_valid_loss = valid_loss
                torch.save(model.state_dict(), "mask_nomask_next.pt")

            end_time = time.monotonic()

            epoch_mins, epoch_secs = epoch_time(start_time, end_time)
            print(f"Epoch: {epoch+1:02} | Epoch Time: {epoch_mins}m {epoch_secs}s")
            print(f"\tTrain Loss: {train_loss:.3f} | Train Acc @1: {train_acc_1*100:6.2f}%")
            print(f"\tValid Loss: {valid_loss:.3f} | Valid Acc @1: {valid_acc_1*100:6.2f}%")
    

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [X] `area/tracking`: Tracking Service, tracking client APIs, auto logging

Interface 
- [X] `area/uiux`: Front-end, user experience, JavaScript, plotting

**Comment by dmatrix** [comment]
@abhiksark Note that PyTorch autolog only works for [PyTorch Lightning models](https://mlflow.org/docs/latest/python_api/mlflow.pytorch.html#mlflow.pytorch.autolog), not for all PyTorch models. So no surprise that it does not work for general PyTorch models.

So this is not effectively a bug, but a feature enhancement.

**Comment by abhiksark** [comment]
Hi, @dmatrix thanks for updating me on this!

**Comment by dmatrix** [comment]
cc: @harupy Do you know if this is on the roadmap?

**Comment by harupy** [comment]
@dmatrix I don't think this is on the roadmap.

**Comment by bdzyubak** [comment]
@dmatrix In my case, I did make a pytorch lightning module but it is still not logging anything. Do I have to add mlflow.log, even when using autologger? 
entrypoint: https://github.com/bdzyubak/torch-control/blob/main/projects/NaturalLanguageProcessing/MovieReviewAnalysis/fine_tune_on_kaggle_movie_sentiment.py
module: https://github.com/bdzyubak/utils/blob/00f3d439bf32aeaff50bda9387b92055a063ef62/LLM_pytorch_lighting_wrapper.py
